src/main/java/com/sprint/mission/discodeit/service/basic/BasicUserService.java

```diff
@@ -0,0 +1,172 @@
+package com.sprint.mission.discodeit.service.basic;
+
+import com.sprint.mission.discodeit.dto.binaryContent.BinaryContentCreateRequest;
+import com.sprint.mission.discodeit.dto.user.UserCreateRequest;
+import com.sprint.mission.discodeit.dto.user.UserResponse;
+import com.sprint.mission.discodeit.dto.user.UserUpdateRequest;
+import com.sprint.mission.discodeit.entity.BinaryContent;
+import com.sprint.mission.discodeit.entity.User;
+import com.sprint.mission.discodeit.entity.UserStatus;
+import com.sprint.mission.discodeit.repository.BinaryContentRepository;
+import com.sprint.mission.discodeit.repository.UserRepository;
+import com.sprint.mission.discodeit.repository.UserStatusRepository;
+import com.sprint.mission.discodeit.service.UserService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+@Service
+@RequiredArgsConstructor
+public class BasicUserService implements UserService {
+    private final UserRepository userRepository;
+    private final UserStatusRepository userStatusRepository;
+    private final BinaryContentRepository binaryContentRepository;
+
+        private void checkDuplicateUsername(String username) {
+        User existingUser = userRepository.findByUsername(username);
+        if (existingUser != null) {
+            throw new IllegalArgumentException("이미 존재하는 사용자 이름입니다: " + username);
+        }
+    }
+
+    private void checkDuplicateEmail(String email) {
+        List<User> allUsers = userRepository.findAll();
+        for (User user : allUsers) {
+            if (user.getEmail().equals(email)) {
+                throw new IllegalArgumentException("이미 존재하는 이메일입니다: " + email);
+            }
+        }
+    }
+
+    @Override
+    public UserResponse createUser(UserCreateRequest request) {
+        checkDuplicateUsername(request.username());
+        checkDuplicateEmail(request.email());
+
+        User user = new User(request.username(), request.email(), request.password());
+        userRepository.save(user);
+
+        UserStatus userStatus = new UserStatus(user.getId());
+        userStatusRepository.save(userStatus);
+
+        return UserResponse.from(user, userStatus);
```

Comment:
정적 팩토리 메서드를 잘 사용해주셨네요 💯   
다만, 인자가 2개 이상일 땐 보통 of를 많이 사용하기에 of로 바꿔주면 좋을 것 같아요.   

```suggestion
        return UserResponse.of(user, userStatus);
```

from은 인자가 1개일 때 보통 사용해요!